### PR TITLE
Fix policy entropy

### PR DESCRIPTION
**Pull request by CatherineSue**
This commit fixes policy entropy. Currently it uses the following three
options:
• maximum entropy (augmented reward)
• with a logli estimator (-logli(action|policy)
• with the gradient turned off (tf.stop_gradient(entropy))

See: #281

### CODE REVIEW

File: garage/tf/algos/npo.py [hunk 1]
```diff
@@ -358,22 +363,37 @@ def _build_entropy_term(self, i):
...
                                         [-1, self.max_path_length])
 
             # This prevents entropy from becoming negative for small policy std
             if self._use_softplus_entropy:
                 policy_entropy = tf.nn.softplus(policy_entropy)
 
-            policy_entropy = tf.reduce_mean(policy_entropy * i.valid_var)
+            policy_entropy = policy_entropy * i.valid_var
+
+            if self._stop_entropy_gradients:
```
Comment (ryanjulian) [inline garage/tf/algos/npo.py]: stop_entropy_gradient

File: garage/tf/algos/npo.py [hunk 2]
```diff
@@ -36,10 +36,14 @@ def __init__(self,
                  policy=None,
                  policy_ent_coeff=1e-2,
                  use_softplus_entropy=True,
+                 log_estimate_entropy=True,
```
Comment (ryanjulian) [inline garage/tf/algos/npo.py]: use_logli_entropy

File: garage/tf/algos/npo.py [hunk 3]
```diff
@@ -36,10 +36,14 @@ def __init__(self,
                  policy=None,
                  policy_ent_coeff=1e-2,
                  use_softplus_entropy=True,
```
Comment (ryanjulian) [inline garage/tf/algos/npo.py]: is this still needed?

File: garage/tf/algos/npo.py [hunk 4]
```diff
@@ -358,22 +363,37 @@ def _build_entropy_term(self, i):
...
                     i.flat.obs_var,
                     i.flat.policy_state_info_vars,
                     name="policy_dist_info_flat")
+                policy_log_likeli_flat = self.policy.distribution.log_likelihood_sym(  # noqa: E501
+                    i.flat.action_var,
+                    policy_dist_info_flat,
+                    name="policy_log_likeli")
+
+            if self._use_logli_entropy:
+                policy_entropy_flat = policy_log_likeli_flat
```
Comment (zhanpenghe) [inline garage/tf/algos/npo.py]: @ryanjulian Shouldn't this be negative log likelihood if using it for entropy?

File: garage/tf/algos/npo.py (hunk 3 above)
Comment (CatherineSue) [reply garage/tf/algos/npo.py]: Yes

File: garage/tf/algos/npo.py (hunk 4 above)
Comment (CatherineSue) [reply garage/tf/algos/npo.py]: Are you suggesting to change the name? 
log_likelihood_sym returns a negative log likelihood.

Comment (zhanpenghe) [reply garage/tf/algos/npo.py]: Oh ok. Then just change the name. I did not notice that it returns negative log likelihood already.

**Comment by ryanjulian** [comment]
Do any of the examples even use entropy?

**Comment by zhanpenghe** [comment]
`policy_ent_coeff` was set as `1e-2` before so any of the examples that does not explicit set it to 0 would use entropy.

**Comment by CatherineSue** [comment]
The default value of `use_logli_entropy` is True, so every related example and benchmark will be using negative log likelihood as entropy term

File: garage/tf/algos/npo.py (hunk 4 above)
Comment (CatherineSue) [reply garage/tf/algos/npo.py]: Done.

**Comment by CatherineSue** [comment]
I changed the default policy_ent_coeff to 0.0.
